modules/core/server/controllers/mongoose-filtering.server.controller.js

```diff
@@ -0,0 +1,54 @@
+﻿// Service for managing Filtering/Parameterized-Queries.
+
+var _query = null;
+var _modifiedQuery = null;
+var _totalDocumentCount = null;
+var _page = 0;
+var _take = 50; // default to prevent unreasonable take() requests
```

Comment:
Different clients might also want different number of pages (mobile), so it
should probably be possible to se it at URL/headers.

On 3 Sep 2016 09:32, "Michael Leanos" <notifications@github.com> wrote:

> In modules/core/server/controllers/mongoose-filtering.server.controller.js
> <https://github.com/meanjs/mean/pull/1263#discussion_r77432291>:
>
> > @@ -0,0 +1,54 @@
> > +﻿// Service for managing Filtering/Parameterized-Queries.
> > +
> > +var _query = null;
> > +var _modifiedQuery = null;
> > +var _totalDocumentCount = null;
> > +var _page = 0;
> > +var _take = 50; // default to prevent unreasonable take() requests
>
> I think I should move the default, and cap (below) for _take
> responsibility to the collection controller. We may want have different
> defaults & caps on each collection. For instance, a messaging module might
> want only the last 50 messages returned, and Articles may want 50.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/meanjs/mean/pull/1263/files/6e0a72716b94833accb9db55cb0a8b742f758b1d#r77432291>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFUgN2v4IOGXRAr4U5hDkkgCHYfxEuSks5qmRSEgaJpZM4HvPtf>
> .
>